server/sublist.go

```diff
@@ -83,7 +83,7 @@ func NewSublist() *Sublist {
 // Insert adds a subscription into the sublist
 func (s *Sublist) Insert(sub *subscription) error {
 	// copy the subject since we hold this and this might be part of a large byte slice.
-	subject := string(sub.subject)
+	subject := (" " + string(sub.subject))[1:]
```

Comment:
Yeah, I'm also still not happy with the current solution. I still want to try some other things, so this should probably wait till we have something better.